engine/src/main/java/org/terasology/engine/world/block/regions/ActAsBlockComponent.java

```diff
@@ -3,14 +3,20 @@
 package org.terasology.engine.world.block.regions;
 
 
-import org.terasology.engine.entitySystem.Component;
 import org.terasology.engine.world.block.family.BlockFamily;
+import org.terasology.gestalt.entitysystem.component.Component;
 
 /**
  * An entity with this component will act as that block - producing block style damage effects, take damage as that block would.
  *
  */
-public final class ActAsBlockComponent implements Component {
+public final class ActAsBlockComponent implements Component<ActAsBlockComponent> {
     public BlockFamily block;
     public boolean dropBlocksInRegion;
+
+    @Override
+    public void copy(ActAsBlockComponent other) {
+        this.block = other.block;
```

Comment:
Do we need to deep-copy `BlockFamily`?
And why is that member called "block" if it's a block family :roll_eyes: